### ISSUE TITLE
Steam Universe changing upon input (web panel)

### ISSUE THREAD

**Issue by sneak-it** [sbpp/sourcebans-pp#568]
What are the steps to reproduce this issue?
-------------------------------------------
1. Enter a ban into the web panel, using the 1 steam universe (STEAM_1:X:XXXXXX)
Alternatively:
1. Edit a ban (including from a gameserver) in any way, and update ban.

What happens?
-------------
The Steam ID gets entered as a 0 universe SteamID (STEAM_0:X:XXXXXX).

What were you expecting to happen?
----------------------------------
The Steam ID to be entered with the 1 universe.

Any logs, error output, etc.?
----------------------------

N/A

Any other comments?
-------------------
I prefer to add all Steam ID's as the 1 universe, as all my gameserver bans come through as 1. Searching numerous bans from the same user is split up between STEAM_0 and STEAM_1, if the user has been banned from both web and directly from in-game.

What versions of software are you using?
----------------------------------------
**Operating System:** Ubuntu 16.04
**SourceBans++ Version:** 1.6.3 Release
**PHP Version:** 7.2
**MySQL Version:** 5.7
**Link to your project:** https://snksrv.com/bans
**Link to a phpinfo() output:** PM if this is necessary.

**Comment by borzaka** [comment]
I can confirm this issue;
Editing a Bans/Comms block will change the Steam ID universe to 0

For example I have this cheater in CS:GO:
`L 10/07/2019 - 17:51:18: "k0b4Y<444><STEAM_1:0:429025023><>" connected, address ""`

I manually banned him, later edited and added comment and then the Steam ID changed to `STEAM_0:0:429025023` and I was unable to change it back to the correct `STEAM_1:0:429025023`

But strange, if you look up his ID in STEAMID, it says that the `STEAM_0:0` is the correct one:
https://steamid.io/lookup/76561198818315774

Another example, where Steam ID changed to 0 Universe:
`L 10/07/2019 - 17:52:08: "pl4tYpus<445><STEAM_1:0:28972761><>" connected, address ""`

`L 10/07/2019 - 17:55:49: [sbpp_main.smx] "Console<0><Console><Console>" banned "pl4tYpus<446><STEAM_1:0:28972761><>" (minutes "0") (reason "SMAC 0.8.6.6: Aimbot Detected")
`

Web panel after edit:
![image](https://user-images.githubusercontent.com/1230402/66351711-66a81580-e95e-11e9-9197-b9b5758c7f45.png)

When trying to add a new with `STEAM_1:0:28972761`, it says `SteamID: STEAM_0:0:28972761 is already banned. `

So, I don't understand what is happening and why :)
But I'am afraid that my bans will not work :(

Plz, try to fix this first if you can, and then move to another Steam ID format.

**Comment by borzaka** [comment]
Another use case where the Steam ID changed:
Banned a player ingame, and then edited the ban on the Web Panel. The Steam ID changed from `STEAM_1:xxx` to `STEAM_0:xxx`

So my ban is worthless? :( I can't change back to `STEAM_1:xxx`

**Comment by rumblefrog** [comment]
You can run update queries directly on the table.

**Comment by borzaka** [comment]
Yes, I could, and I could also manage everything from phpMyAdmin, but it's not convenient :)
And sometimes I don't pay attention which ban was with 1 universe.
But why Web Panel changing an already inserted Steam ID? It sounds like an easy fix.

**Comment by CrazyHackGUT** [comment]
Because first number in SteamID is not affect and web-panel tries keep everything in one form?

**Comment by borzaka** [comment]
So my bans are safe? The cheater will not come back after his Steam ID changed from `STEAM_1:xxx` to `STEAM_0:xxx` during a ban update in the Web Panel?

**Comment by rumblefrog** [comment]
Yes, it should be safe as the query does not check against the universe.

Edit: Well it works somewhat well, it's not to say it's without its performance issues as the table gets increasingly larger. Since the regexp query cannot utilize the index, it must perform a full-table scan to get a result.

**Comment by borzaka** [comment]
Maybe I have found an additional info to this issue. I'am using only CS:GO servers for my SourceBans++ install, which has this setting:
![image](https://user-images.githubusercontent.com/1230402/67060794-84475d00-f15e-11e9-9dc0-6285837881f5.png)

So the SteamID Universe for a CS:GO banned user should stay at 1 if it was 1 after any edit, right?

**Comment by rumblefrog** [comment]
Yes and no, some parts of the panel do not take account about that setting.

**Comment by borzaka** [comment]
I have another riddle for you in this Steam Universe issue :) Can't kick a `1` Universe user:

```
# userid name uniqueid connected ping loss state rate adr
# 147 9 "owning nns" STEAM_1:0:1856778 02:54 32 0 active 786432 xx.xx.xx.xx:54301
L 10/16/2019 - 18:35:01: rcon from "x.x.x.x:43964": command "status"
uniqueid "STEAM_0:0:1856778" not found
L 10/16/2019 - 18:35:03: rcon from "x.x.x.x:43964": command "kickid STEAM_0:0:1856778 "You have been banned by this server, check http://hostname/sourcebans/index.php for more info.""
```

The Web Panel says he's not on the server anymore, or got kicked, but actually not.

Problem: `1` is changed to `0`, just like during editing.

**Comment by rumblefrog** [comment]
Yes, this is also known.

**Comment by borzaka** [comment]
The kick message got fixed in 6d53eeb5b4069d087c7483c345fbb75fb54c1e16, big up for that!

**Comment by rumblefrog** [comment]
Let's avoid using issues as off-topic discussion.
